src/libarpack.jl

```diff
@@ -152,17 +152,20 @@ function eupd_wrapper(T, n::Integer, sym::Bool, cmplx::Bool, bmat,
     select = Vector{BlasInt}(undef, ncv)
     info   = Ref{BlasInt}(0)
 
-    dmap = x -> abs.(x)
+    dmap = if which == "LM" || which == "SM"
+      abs
+    elseif which == "LR" || which == "LA" || which == "BE" || which == "SR" || which == "SA"
+      real
+    elseif which == "LI" || which == "SI"
+      imag
+    else
+      error("unknown which string $which")
+    end
+
+    rev = which[1] == 'L'
+
```

Comment:
This level of the arpack call is very tricky. @tomhaber why you prefer your version of `dmap` combined with `rev`? Over just `dmap` including a sign in previous version?

Reply:
For me, it simplifies the logic. 
If I remember correctly, this version allows shift-invert to also work with non magnitude modes and fixes wrong ordering for SM and no shift-invert.